File: scripts/conformance-eval/fixture/src/domain/result.ts
export type Result<T, E> = { readonly ok: true; readonly value: T } | { readonly ok: false; readonly error: E };
export const ok = <T>(value: T): { ok: true; value: T } => ({ ok: true, value });
export const err = <E>(error: E): { ok: false; error: E } => ({ ok: false, error });

File: scripts/review-eval/base-java/src/main/java/com/example/app/domain/Result.java
// atelier Java asset: the sealed Result union (hard rule 16, Java expression).
// Copy into your domain package; rename `com.example.app.domain` to your own.
// Ok/Err complete the union; use-cases pattern-match with an exhaustive switch.
package com.example.app.domain;

public sealed interface Result<T, E> permits Ok, Err {}
